app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -1227,7 +1227,11 @@ public ArtifactSearchResultsDto searchArtifacts(Set<SearchFilter> filters, Order
             orderByQuery.append(orderDirection.name());
 
             // Add limit and offset to artifact query
-            limitOffset.append(" LIMIT ? OFFSET ?");
+            if ("sqlserver".equals(sqlStatements.dbType())) {
+                limitOffset.append(" OFFSET ? ROWS FETCH NEXT ? ROWS ONLY");
```

Comment:
would you mind adding a comment regarding this limitation? maybe with a comment regarding where the different syntax comes from.

Reply:
OFFSET ROWS FETCH NEXT is the equivalent of LIMIT OFFSET in postgres: https://learn.microsoft.com/en-us/sql/t-sql/queries/select-order-by-clause-transact-sql?view=sql-server-ver15#syntax

Reply:
and you can see there's a swap in the arguments, which triggers this ugly if-else